### ISSUE TITLE
Support for elemMatch isn't consistent with Mongo

### ISSUE THREAD

**Issue by redgeoff** [kofrasa/mingo#163]
Thanks again for the awesome library @kofrasa!

I'm putting the finishing touches on swapping out siftjs with mingo in [MSON](https://github.com/redgeoff/mson) so that MSON can also support aggregations.

In doing so, I found that there appears to be an issue with how mingo is handling `$elemMatch`. 

Specifically, the following code is not finding a match:

```
import mingo from "mingo";

const collection = [
  {
    value: [
      {
        id: "1",
        color: "red"
      },
      {
        id: "2",
        color: "red"
      }
    ]
  }
];

const criteria = {
  value: {
    $elemMatch: {
      $and: [
        {
          id: {
            $ne: "1"
          }
        },
        {
          color: "red"
        }
      ]
    }
  }
};

const query = new mingo.Query(criteria);
const cursor = query.find(collection);
console.log(cursor.all());
```

Here is a [codesandbox](https://codesandbox.io/s/mingo-elemmatch-536jo?file=/src/index.js) for this example and here is a [similar example](https://mongoplayground.net/p/F9z8CTyQACR) **working** on an actual MongoDB instance.

Would you like me to take a closer look and see if I can create a PR?

**Comment by redgeoff** [comment]
I'm digging in a bit deeper and it appears that the issue is with Mingo's handling of `$and` when nested in a `$elemMatch`. I was able to find a workaround for this particular case by using the following query instead:
```
const criteria = {
  value: {
    $elemMatch: {
      id: {
        $ne: "1"
      },
      color: "red"
    }
  }
};
```

This only works though as the `$and` operation is implicit. If for example, we needed to use the `$or` operator like the following then it breaks:
```
const criteria = {
  value: {
    $elemMatch: {
      $or: [
        {
          id: "1"
        },
        {
          color: "red"
        }
      ]
    }
  }
};
```

I'll dig in a bit deeper and see if I can craft a PR. One thing that would be helpful is some mention of how to run a specific mingo test. I hacked things up a bit to get it to work, but I'm guessing there is probably a better way that doesn't require changing code.
